### ISSUE TITLE
Pangesture does not trigger exit/completed/cancel event while pan moves outside of the layout/control

### ISSUE THREAD

**Issue by Sasikumar3595** [dotnet/maui#5191]
### Description

We have created a custom layout to hold scrollable content like paging and the custom layout has a pan gesture to swipe between its children. When swipe the custom layout and move outside the custom layout then the swipe is not completed/canceled so the UI is not rendered correctly, it shows 1st page half and 2nd page half. Guide us to resolve the issue.

### Steps to Reproduce

1. Create a custom layout with pan gesture
2. Swipe inside the layout to trigger a pan updated event.
3. Start pan inside the layout and move outside the custom layout then it does not trigger the pan canceled/completed

### Version with bug

Preview 13 (current)

### Last version that worked well

Preview 13 (current)

### Affected platforms

iOS, Android, Windows, macOS

### Affected platform versions

Windows 10.0.17134.0 iOS 15 Android 30 

### Did you find any workaround?

No

### Relevant log output

```shell
Not Applicable
```

**Comment by jfversluis** [comment]
Could you please add some relevant code or, even better, a reproduction sample that shows this issue? That will greatly speed up the process of diagnosing and fixing a potential bug. Thanks!

**Comment by Sasikumar3595** [comment]
@jfversluis ,

Our architecture is the grid contains two-column (one (Green background) is a grid with pan gesture and another one (Red background) is scroll view).

We are using the grid pan gesture on left child (Green background) to move to right side child (Red background) but the pan gesture completed/canceled event not triggered.

```
        Grid customGrid = new Grid();
        StackLayout customStckLayout = new StackLayout();
        for(int i =0; i < 100; i++)
        {
            customStckLayout.Add(new Grid()
            {
                BackgroundColor = Colors.Red,
                Margin = new Thickness(1),
                HeightRequest = 50,
            });
        }
        ScrollView customScrollView = new ScrollView();
        customScrollView.Orientation = ScrollOrientation.Vertical;
        customScrollView.Content = customStckLayout;
        customGrid.ColumnDefinitions = new ColumnDefinitionCollection {
    new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) },
     new ColumnDefinition { Width = new GridLength(1, GridUnitType.Star) }
            };
        Grid.SetColumn(customScrollView, 1);
        customGrid.Add(customScrollView);
        Grid customGrid2 = new Grid();
        customGrid2.BackgroundColor = Colors.Green;
        Grid.SetColumn(customGrid2, 0);
        customGrid.Add(customGrid2);
        PanGestureRecognizer pan = new PanGestureRecognizer();
        pan.PanUpdated += Pan_PanUpdated;
        customGrid2.GestureRecognizers.Add(pan);
       this.Content = customGrid;

    private void Pan_PanUpdated(object sender, PanUpdatedEventArgs e)
    {
        System.Diagnostics.Debug.WriteLine("Panned -- " + e.StatusType.ToString());
    }

```
![Issue](https://user-images.githubusercontent.com/44917578/157683982-a76423a3-2d74-48ef-a634-d24e727bf3b9.png)

**Comment by PineYi** [comment]
Verified Repro with Windows. Repro Project is available:
[#5191.zip](https://github.com/dotnet/maui/files/8228387/5191.zip)

**Comment by espenrl** [comment]
I'm having the same problem using PanGestureRecognizer on Windows. If I move the mouse fast enough such that it escapes the view (which the PanGestureRecognizer is attached to) there is never another PanUpdated event. I would expect the Cancelled status to be raised.

**Comment by homeyf** [comment]
Verified this issue with Visual Studio Enterprise 17.6.0 Preview 2.0. Repro on Android Emulator with above project.
![image](https://user-images.githubusercontent.com/82450723/233007136-610231bc-c31f-4ac3-a639-2f89a32698b6.png)

**Comment by hartez** [comment]
This is not how pan gestures work. Moving outside of the view bounds does not fire the "Completed" event - only lifting the touch will do that. This is by design. 

Are you are attempting to implement a "drag and drop" scenario? If so, then you should follow the [drag and drop documentation](https://learn.microsoft.com/en-us/dotnet/maui/fundamentals/gestures/drag-and-drop). 

If not, why do you want the pan gesture to be cancelled/completed when the touch exits the view bounds? What are you trying to accomplish?

**Comment by Sasikumar3595** [comment]
@hartez 

I did not get the completed/cancel event while lifting the touch, I cannot identify the pan was completed/ended.

I was trying to implement the **custom swipeable** layout based on the index value.

**Comment by Sasikumar3595** [comment]
@mattleibow 

No, I did not receive the touch event when the finger was lifted, so I cannot determine if the pan was completed/ended

**Comment by jsuarezruiz** [comment]
The issue will be fixed by https://github.com/dotnet/maui/pull/15700
